### PR TITLE
Include recyclable memorystream in new NOTICE.txt

### PR DESCRIPTION
**Pull request by Mpdreamz**

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -0,0 +1,31 @@
+Elasticsearch .NET Transport
+Copyright 2012-2018 Elasticsearch B.V.
```
Comment (gtback) [inline NOTICE.txt:2]: We'll want to update these dates to include any years where significant (copyrightable) changes were made to this project.

File: src/Elastic.Transport/Components/Providers/RecyclableMemoryStream.cs [hunk 2]
```diff
@@ -1,7 +1,3 @@
-// Licensed to Elasticsearch B.V under one or more agreements.
```
Comment (gtback) [inline src/Elastic.Transport/Components/Providers/RecyclableMemoryStream.cs:1]: What is the reason for removing the header from this and the other files?

Comment (Mpdreamz) [reply src/Elastic.Transport/Components/Providers/RecyclableMemoryStream.cs:1]: These contain no modifications from us, happy to add these back if needed!

File: NOTICE.txt (hunk 1 above)
Comment (stevejgordon) [reply NOTICE.txt]: I've updated the end date on this.

File: src/Elastic.Transport/Components/Providers/RecyclableMemoryStream.cs (hunk 2 above)
Comment (stevejgordon) [reply src/Elastic.Transport/Components/Providers/RecyclableMemoryStream.cs]: Added these back in for all files.

**Pull request merged**